frontend/src/features/recommendation/LanguageFilter.tsx

```diff
@@ -1,37 +1,50 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
-import { ChoicesFilterSection } from 'src/components';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
 import {
   availableRecommendationsLanguages,
   getLanguageName,
-} from 'src/utils/constants';
+} from 'src/utils/recommendationsLanguages';
+import { TitledSection } from 'src/components';
 
 interface Props {
   value: string;
   onChange: (value: string) => void;
 }
 
-function LanguageFilter(props: Props) {
+function LanguageFilter({ value, onChange }: Props) {
   const { t } = useTranslation();
 
-  const languageChoices: Record<string, string> = useMemo(
-    () =>
-      Object.fromEntries(
-        availableRecommendationsLanguages.map((language) => [
-          language,
-          getLanguageName(t, language),
-        ])
-      ),
+  const getOptionLabel = useCallback(
+    (option) => getLanguageName(t, option),
     [t]
   );
 
+  const handleChange = useCallback(
+    (event: React.SyntheticEvent<Element, Event>, newValue: string[]) => {
+      onChange(newValue.join(','));
+    },
+    [onChange]
+  );
+
+  const arrayValue = useMemo(
+    () => (value === '' ? [] : value.split(',')),
+    [value]
+  );
+
   return (
-    <ChoicesFilterSection
-      title={t('filter.language')}
-      multipleChoice
-      choices={languageChoices}
-      {...props}
-    ></ChoicesFilterSection>
+    <TitledSection title={t('filter.language')}>
+      <Autocomplete
+        multiple
+        options={availableRecommendationsLanguages}
```

Comment:
The order of these options may look random and not be very practical. 
Maybe we could sort the labels by alphabetical order?
![image](https://user-images.githubusercontent.com/4726554/158177974-8ce21706-e838-4e5b-a580-036364b61de2.png)